virttest/utils_qemu.py

```diff
@@ -19,10 +33,107 @@ def get_qemu_version(bin_path):
     :raise OSError: If unable to get that
     :return: A tuple of normalized version and package version
     """
-    output = process.system_output("%s -version" % bin_path,
-                                   verbose=False,
-                                   ignore_status=True).decode()
+    output = _get_info(bin_path, "-version")
     matches = QEMU_VERSION_RE.match(output)
     if matches is None:
         raise OSError('Unable to get the version of qemu')
     return matches.groups()
+
+
+def get_machines_info(bin_path):
+    """
+    Return all machines information supported by qemu
+
+    :param bin_path: Path to qemu binary
+    :return: A dict of all machines
+    """
+    output = _get_info(bin_path, r"-machine help", allow_output_check="combined")
+    machines = re.findall(r"^([a-z][0-9a-zA-Z._-]+)\s+(.*)$", output, re.M)
```

Comment:
I'd recommend using `re.split(r"\s+", l, 1)`

Reply:
The machine help info is as follows:
```
# /usr/libexec/qemu-kvm -machine help
Supported machines are:
none                 empty machine
pseries-rhel7.2.0    pSeries Logical Partition (PAPR compliant)
pseries-rhel7.3.0    pSeries Logical Partition (PAPR compliant)
pseries-rhel7.3.0-sxxm pSeries Logical Partition (PAPR compliant)
pseries-rhel7.4.0    pSeries Logical Partition (PAPR compliant)
pseries-rhel7.4.0-sxxm pSeries Logical Partition (PAPR compliant)
pseries-rhel7.5.0    pSeries Logical Partition (PAPR compliant)
pseries-rhel7.5.0-sxxm pSeries Logical Partition (PAPR compliant)
pseries              pSeries Logical Partition (PAPR compliant) (alias of pseries-rhel7.6.0)
pseries-rhel7.6.0    pSeries Logical Partition (PAPR compliant) (default)
pseries-rhel7.6.0-sxxm pSeries Logical Partition (PAPR compliant)
```
I have to ignore the line 1, if want to use `re.split/str.split`, need to match them line by line.
It looks more succinct now, what do you think, @luckyh